twiliochat/ChannelManager.swift

```diff
@@ -85,54 +85,54 @@ class ChannelManager: NSObject {
   func sortChannels() {
     let sortSelector = #selector(NSString.localizedCaseInsensitiveCompare(_:))
     let descriptor = NSSortDescriptor(key: "friendlyName", ascending: true, selector: sortSelector)
-    channels!.sortUsingDescriptors([descriptor])
+    channels!.sort(using: [descriptor])
   }
 
   // MARK: - Create channel
 
-  func createChannelWithName(name: String, completion: (Bool, TWMChannel?) -> Void) {
+  func createChannelWithName(name: String, completion: @escaping (Bool, TWMChannel?) -> Void) {
     if (name == ChannelManager.defaultChannelName) {
       completion(false, nil)
       return
     }
 
     let channelOptions:[NSObject : AnyObject] = [
-      TWMChannelOptionFriendlyName: name, TWMChannelOptionType: TWMChannelType.Public.rawValue
+      TWMChannelOptionFriendlyName as NSObject: name as AnyObject, TWMChannelOptionType as NSObject: TWMChannelType.public.rawValue as AnyObject
     ]
-    UIApplication.sharedApplication().networkActivityIndicatorVisible = true;
-    self.channelsList?.createChannelWithOptions(channelOptions) { result, channel in
-      UIApplication.sharedApplication().networkActivityIndicatorVisible = false;
-      completion(result.isSuccessful(), channel)
+    UIApplication.shared.isNetworkActivityIndicatorVisible = true;
+    self.channelsList?.createChannel(options: channelOptions) { result, channel in
+      UIApplication.shared.isNetworkActivityIndicatorVisible = false;
+      completion((result?.isSuccessful())!, channel)
     }
   }
 }
 
 // MARK: - TwilioIPMessagingClientDelegate
 extension ChannelManager : TwilioIPMessagingClientDelegate {
-  func ipMessagingClient(client: TwilioIPMessagingClient!, channelAdded channel: TWMChannel!) {
-    dispatch_async(dispatch_get_main_queue(), {
+  func ipMessagingClient(_ client: TwilioIPMessagingClient!, channelAdded channel: TWMChannel!) {
+    DispatchQueue.main.async {
```

Comment:
Perhaps not important in this demo but could this be done w/ `DispatchQueue.global.async` given that these are network tasks?